systems/src/overhead/mod.rs

```diff
@@ -0,0 +1,235 @@
+            state: OnOffPushButtonState::On,
+            fault: false,
+            available: false,
+        }
+    }
+
+    pub fn new_off() -> OnOffPushButton {
+        OnOffPushButton {
+            state: OnOffPushButtonState::Off,
+            fault: false,
+            available: false,
+        }
+    }
+
+    pub fn set(&mut self, value: bool) {
+        self.state = if value {
+            OnOffPushButtonState::On
+        } else {
+            OnOffPushButtonState::Off
+        };
+    }
+
+    pub fn set_fault(&mut self, fault: bool) {
+        self.fault = fault;
+    }
+
+    pub fn turn_on(&mut self) {
+        self.state = OnOffPushButtonState::On;
+    }
+
+    pub fn turn_off(&mut self) {
+        self.state = OnOffPushButtonState::Off;
+    }
+
+    pub fn set_available(&mut self, available: bool) {
+        self.available = available;
+    }
+
+    pub fn shows_available(&self) -> bool {
+        self.available
+    }
+
+    pub fn has_fault(&self) -> bool {
+        self.fault
+    }
+
+    pub fn is_on(&self) -> bool {
+        self.state == OnOffPushButtonState::On
+    }
+
+    pub fn is_off(&self) -> bool {
+        self.state == OnOffPushButtonState::Off
+    }
+}
+
+#[derive(PartialEq)]
+pub enum NormalAltnPushButtonState {
+    Normal,
+    Altn,
+}
+
+pub struct NormalAltnPushButton {
+    state: NormalAltnPushButtonState,
+    fault: bool,
+}
+impl NormalAltnPushButton {
+    pub fn new_normal() -> NormalAltnPushButton {
+        NormalAltnPushButton {
+            state: NormalAltnPushButtonState::Normal,
+            fault: false,
+        }
+    }
+
+    pub fn new_altn() -> NormalAltnPushButton {
+        NormalAltnPushButton {
+            state: NormalAltnPushButtonState::Altn,
+            fault: false,
+        }
+    }
+
+    pub fn push_altn(&mut self) {
+        self.state = NormalAltnPushButtonState::Altn;
+    }
+
+    pub fn is_normal(&self) -> bool {
+        self.state == NormalAltnPushButtonState::Normal
+    }
+
+    pub fn is_altn(&self) -> bool {
+        self.state == NormalAltnPushButtonState::Altn
+    }
+}
+
+#[derive(PartialEq)]
+pub enum AutoOffPushButtonState {
+    Auto,
+    Off,
+}
+
+pub struct AutoOffPushButton {
+    state: AutoOffPushButtonState,
+    fault: bool,
+}
+impl AutoOffPushButton {
+    pub fn new_auto() -> AutoOffPushButton {
+        AutoOffPushButton {
+            state: AutoOffPushButtonState::Auto,
+            fault: false,
+        }
+    }
+
+    pub fn new_off() -> AutoOffPushButton {
+        AutoOffPushButton {
+            state: AutoOffPushButtonState::Off,
+            fault: false,
+        }
+    }
+
+    pub fn push_off(&mut self) {
+        self.state = AutoOffPushButtonState::Off;
+    }
+
+    pub fn is_auto(&self) -> bool {
+        self.state == AutoOffPushButtonState::Auto
+    }
+
+    pub fn is_off(&self) -> bool {
+        self.state == AutoOffPushButtonState::Off
+    }
+}
+
+pub struct FirePushButton {
+    released: bool,
```

Comment:
`released` sounds a tiny bit ambiguous. Is this the usual terminology? Do pilots "release" the fire push buttons?

Reply:
This is the terminology used in the FCOM:

> When the flight crew pushes it, the pushbutton is released